### PR TITLE
[3822] Tied NewsletterSubscriptionContainer to router (recreated)

### PR DESCRIPTION
**Pull request by S-Arturs**
**Related issue(s):**
* Fixes https://github.com/scandipwa/scandipwa/issues/3822

**Problem:**
* Newsletter subscription error won't go away when user changes pages

**In this PR:**
* Tied NewsletterSubscriptionContainer to react router so it refreshes on every page change

### CODE REVIEW

File: packages/scandipwa/src/component/NewsletterSubscription/NewsletterSubscription.container.js [hunk 1]
```diff
@@ -97,12 +98,13 @@ export class NewsletterSubscriptionContainer extends PureComponent {
...
               { ...this.containerProps() }
               { ...this.containerFunctions }
+              key={ window.location.pathname }
             />
         );
     }
 }
 
-export default connect(
+export default withRouter(connect(
```
Comment (tatiana-scandi) [inline packages/scandipwa/src/component/NewsletterSubscription/NewsletterSubscription.container.js:107]: If you take pathname from window, there is no need to wrap container into `withRouter` (withRouter will provide access to such props as `history` and `location`, but I don't see that them to be used in this case).

**Review by tatiana-scandi — CHANGES_REQUESTED** [review]
CR failed, needs changes

File: packages/scandipwa/src/component/NewsletterSubscription/NewsletterSubscription.container.js (hunk 1 above)
Comment (S-Arturs) [reply packages/scandipwa/src/component/NewsletterSubscription/NewsletterSubscription.container.js:107]: I tried it, the solution seems not to work without withRouter. I think it has something to do with passing new props every time user changes page, which then refreshes the component.

Comment (tatiana-scandi) [reply packages/scandipwa/src/component/NewsletterSubscription/NewsletterSubscription.container.js:107]: It happens due to no prop being updated, if `withRouter` is not used => nothing to trigger update in component. In this case it's better to use `withRouter` + replace `window.location.pathname` with `this.props.location.pathname` (with props destructuring).

**Review by AleksandrsKondratjevs — APPROVED** [review]

**Pull request merged**